File: ambeth/Ambeth.Security/ambeth/privilege/IPrivilegeProvider.cs
using System;
using System.Collections.Generic;
using De.Osthus.Ambeth.Annotation;
using De.Osthus.Ambeth.Merge.Model;
using De.Osthus.Ambeth.Model;
using De.Osthus.Ambeth.Privilege;
using De.Osthus.Ambeth.Privilege.Model;
using De.Osthus.Ambeth.Util;

namespace De.Osthus.Ambeth.Privilege
{
    public interface IPrivilegeProvider
    {
        IPrivilege GetPrivilegeByObjRef(IObjRef objRef, params ISecurityScope[] securityScopes);

        /// <summary>
        /// Result correlates by-index with the given objRefs
        /// </summary>
        /// <typeparam name="V"></typeparam>
        /// <param name="entities"></param>
        /// <param name="securityScopes"></param>
        /// <returns></returns>
		IPrivilegeResult GetPrivileges<V>(IList<V> entities, params ISecurityScope[] securityScopes);

        /// <summary>
        /// Result correlates by-index with the given objRefs
        /// </summary>
        /// <typeparam name="V"></typeparam>
        /// <param name="objRefs"></param>
        /// <param name="securityScopes"></param>
        /// <returns></returns>
		IPrivilegeResult GetPrivilegesByObjRef<V>(IList<V> objRefs, params ISecurityScope[] securityScopes) where V : IObjRef;

        ITypePrivilege GetPrivilegeByType(Type entityType, params ISecurityScope[] securityScopes);

		ITypePrivilegeResult GetPrivilegesByType(IList<Type> entityTypes, params ISecurityScope[] securityScopes);
    }
}

File: jambeth/jambeth-security/src/main/java/de/osthus/ambeth/privilege/IPrivilegeProvider.java
package de.osthus.ambeth.privilege;

import java.util.List;

import de.osthus.ambeth.merge.model.IObjRef;
import de.osthus.ambeth.privilege.model.IPrivilege;
import de.osthus.ambeth.privilege.model.IPrivilegeResult;
import de.osthus.ambeth.privilege.model.ITypePrivilege;
import de.osthus.ambeth.privilege.model.ITypePrivilegeResult;

public interface IPrivilegeProvider
{
	IPrivilege getPrivilege(Object entity);

	IPrivilege getPrivilegeByObjRef(IObjRef objRef);

	/**
	 * Result correlates by-index with the given objRefs
	 * 
	 * @param objRefs
	 * @param securityScopes
	 * @return
	 */
	IPrivilegeResult getPrivileges(List<?> entities);

	/**
	 * Result correlates by-index with the given objRefs
	 * 
	 * @param objRefs
	 * @param securityScopes
	 * @return
	 */
	IPrivilegeResult getPrivilegesByObjRef(List<? extends IObjRef> objRefs);

	ITypePrivilege getPrivilegeByType(Class<?> entityType);

	ITypePrivilegeResult getPrivilegesByType(List<Class<?>> entityTypes);
}
